test/src/unit-capi-nullable.cc

```diff
@@ -217,35 +207,30 @@ NullableArrayFx::NullableArrayFx() {
 
   tiledb_config_free(&config);
 
+  // Create the UniqueDirectory.
+  temp_dir_.emplace(*vfs_->vfs());
+
   serialize_ = false;
 }
 
 NullableArrayFx::~NullableArrayFx() {
-  remove_dir(FILE_TEMP_DIR);
   tiledb_ctx_free(&ctx_);
   tiledb_vfs_free(&vfs_);
 }
 
-void NullableArrayFx::create_dir(const string& path) {
-  REQUIRE(tiledb_vfs_create_dir(ctx_, vfs_, path.c_str()) == TILEDB_OK);
-}
-
-void NullableArrayFx::remove_dir(const string& path) {
-  int is_dir = 0;
-  REQUIRE(tiledb_vfs_is_dir(ctx_, vfs_, path.c_str(), &is_dir) == TILEDB_OK);
-  if (is_dir)
-    REQUIRE(tiledb_vfs_remove_dir(ctx_, vfs_, path.c_str()) == TILEDB_OK);
-}
-
 void NullableArrayFx::create_array(
-    const string& array_name,
+    const string& array_path,
     const tiledb_array_type_t array_type,
     const vector<test_dim_t>& test_dims,
     const vector<test_attr_t>& test_attrs,
     const tiledb_layout_t cell_order,
     const tiledb_layout_t tile_order) {
-  remove_dir(FILE_TEMP_DIR);
-  create_dir(FILE_TEMP_DIR);
+  // Remove array if it already exists
```

Comment:
If the array already exists then this function should throw. Removal and recreation is more than simply hostile to C.41; it's in open warfare with it.

Each fixture instance has its own unique directory associated with it, and each array for testing should only be written once to that directory. That's what allows automatic cleanup to be reliable. If some tests are creating the same array twice, then the test is bad and needs to be rewritten.

There's only a single test case where this fixture is used, and it reuses the temporary directory 96 times (five loops 3 x 2 x 2 x 2 x 4) for an array with the same name each time. So the test is bad and needs to be rewritten.

It looks like, however, that this is pretty straightforward. The four inner loops are amenable to nested `GENERATE` statements and a `DYNAMIC_SECTION` for the innermost loop body. The currently-outermost (fifth) loop that iterates over attributes could go inside that section.